src/components/UserSelector.tsx

```diff
@@ -1,40 +1,97 @@
-import React from 'react';
+import { useEffect, useRef, useState } from 'react';
+import { User } from '../types/User';
+import classNames from 'classnames';
+
+type Props = {
+  users: User[];
+  setUserSelected: (user: User) => void;
+  userSelected: User | null;
+  handleUserPosts: (user: User) => void;
+  setShowSideBar: (item: number | null) => void;
+};
```

Comment:
You have described the props for this component, but you haven't used PropTypes to validate the props. You should add PropTypes to validate the props passed to the component.